### ISSUE TITLE
no chance to even get a start: flutter_tools depends on webdriver 2.1.2 (Windows 10 latest)

### ISSUE THREAD

**Issue by kklepper** [flutter/flutter#61459]
## Steps to Reproduce

1. installed via `zip` or `git`, no difference with respect to the error 
2. flavor `git` (git clone https://github.com/flutter/flutter.git -b stable): run `flutter_console.bat` in PowerShell
3. run `flutter doctor` or any other command
4. result is: `Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.`

## Logs

```
PS D:\data\programs\dev\flutter> .\flutter_console.bat

        ######## ##       ##     ## ######## ######## ######## ########
        ##       ##       ##     ##    ##       ##    ##       ##     ##
        ##       ##       ##     ##    ##       ##    ##       ##     ##
        ######   ##       ##     ##    ##       ##    ######   ########
        ##       ##       ##     ##    ##       ##    ##       ##   ##
        ##       ##       ##     ##    ##       ##    ##       ##    ##
        ##       ########  #######     ##       ##    ######## ##     ##


 WELCOME to the Flutter Console.
 ===============================================================================

 Use the console below this message to interact with the "flutter" command.
 Run "flutter doctor" to check if your system is ready to run Flutter apps.
 Run "flutter create <app_name>" to create a new Flutter project.

 Run "flutter help" to see all available commands.

 Want to use an IDE to interact with Flutter? https://flutter.dev/ide-setup/

 Want to run the "flutter" command from any Command Prompt or PowerShell window?
 Add Flutter to your PATH: https://flutter.dev/setup-windows/#update-your-path

 ===============================================================================



D:\data\programs\dev\flutter>flutter doctor --verbose
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

D:\data\programs\dev\flutter>flutter --version --verbose
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

PS D:\data\programs\dev\flutter> flutter pub global activate devtools
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

PS D:\data\programs\dev\flutter> flutter channel beta
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)


```

```
```

**Comment by jonahwilliams** [comment]
https://pub.dev/packages/webdriver 2.1.2 definitely exists. Are you using a pub mirror?

**Comment by kklepper** [comment]
@jonahwilliams I'll look into this in a second

As I see from Google `flutter "webdriver 2.1.2"`

```
webdriver | Dart Packagepub.dev › packages › webdriver
webdriver 2.1.2. Published Feb 17, 2020. 3 likes. [unidentified].
```

this seems to be a part of dart which does not seem to be installed with either `Android Studio` or `flutter`. So I installed `dart SDK` with `choco install dart-sdk`, to no avail. From the log:

`2020-07-14 18:17:14,386 18164 [WARN ] -  - dart-sdk - dart-sdk v2.8.4 already installed.`

**Comment by jonahwilliams** [comment]
> this seems to be a part of dart which does not seem to be installed with either Android Studio or flutter. So I installed dart SDK with choco install dart-sdk, to no avail. From the log

Flutter uses its own dart-sdk, installing another one will not affect this.

**Comment by kklepper** [comment]
@jonahwilliams 

https://pub.dev/packages/webdriver tells me to install by one of three methods. Don't understand the first two, `pub install` gives `not found`.

**Comment by kklepper** [comment]
> Flutter uses its own dart-sdk, installing another one will not affect this.

So `flutter` should know, right?

**Comment by jonahwilliams** [comment]
FYI @jonasfj - this might be a pub issue, though I'm not really sure how to start with this one.

**Comment by jonasfj** [comment]
@kklepper are you behind a corporate proxy or similar? (that still shouldn't return 404, telling `pub` that the version doesn't exist).

The best way to tell if it's a `pub` issue, is to reproduce it... and run `pub upgrade --verbose`, maybe by running `flutter-sdk/bin/cache/dart-sdk/bin/pub upgrade --verbose` from within the `flutter-sdk/packages/flutter_tools/` folder.
(assume this will create a dirty sdk folder, so create a fresh one afterwards unless you know how to clean up).

**Comment by kklepper** [comment]
`D:\data\programs\dev\flutter>pub upgrade --verbose` obviously is not correct.

I'd like to do as you tell me, but where do I find `flutter-sdk`?

**Comment by kklepper** [comment]
@jonasfj 

Google `flutter-sdk` just turns me to what I have istalled...

Ok, found it:


```
D:\data\programs\dev\flutter\bin\cache\dart-sdk\bin>pub upgrade --verbose
FINE: Pub 2.8.4
IO  : Spawning "cmd /c ver" in D:\data\programs\dev\flutter\bin\cache\dart-sdk\bin\.
IO  : Finished ver. Exit code 0.
    | stdout:
    | |
    | | Microsoft Windows [Version 10.0.19041.329]
    | Nothing output on stderr.
ERR : Could not find a file named "pubspec.yaml" in "D:\data\programs\dev\flutter\bin\cache\dart-sdk\bin".
FINE: Exception type: FileException
FINE: package:pub/src/pubspec.dart 457:7          new Pubspec.load
    | package:pub/src/package.dart 136:27         new Package.load
    | package:pub/src/entrypoint.dart 156:24      new Entrypoint.current
    | package:pub/src/command.dart 42:59          PubCommand.entrypoint
    | package:pub/src/command/upgrade.dart 48:11  UpgradeCommand.run
    | package:args/command_runner.dart 197:27     CommandRunner.runCommand
    | package:pub/src/command_runner.dart 191:39  PubCommandRunner.runCommand.<fn>
    | dart:async                                  new Future.sync
    | package:pub/src/utils.dart 113:12           captureErrors.wrappedCallback
    | package:stack_trace                         Chain.capture
    | package:pub/src/utils.dart 126:11           captureErrors
    | package:pub/src/command_runner.dart 191:13  PubCommandRunner.runCommand
    | ===== asynchronous gap ===========================
    | dart:async                                  Future.catchError
    | package:pub/src/utils.dart 113:52           captureErrors.wrappedCallback
    | package:stack_trace                         Chain.capture
    | package:pub/src/utils.dart 126:11           captureErrors
    | package:pub/src/command_runner.dart 191:13  PubCommandRunner.runCommand
    | dart:async                                  _completeOnAsyncReturn
    | package:pub/src/command_runner.dart         PubCommandRunner._validatePlatform
    | dart:async                                  _completeOnAsyncReturn
    | package:pub/src/io.dart                     runProcess.<fn>
```

And again:

```
D:\data\programs\dev\flutter>flutter doctor --verbose
Building flutter tool...
Running pub upgrade...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
Error (1): Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)
```

**Comment by jonasfj** [comment]
Yeah, then try running it from the `packages/flutter_tools/` subfolder...

**Comment by kklepper** [comment]
@jonasfj 

Thank you, looks a bit better, but still:

```
D:\data\programs\dev\flutter\packages\flutter_tools>D:\data\programs\dev\flutter\bin\cache\dart-sdk\bin\pub upgrade
Resolving dependencies...
Because flutter_tools depends on webdriver 2.1.2 which doesn't match any versions, version solving failed.
```

**Comment by jonasfj** [comment]
Cool, can you run that with `--verbose`

**Comment by kklepper** [comment]
Ok, good idea, 270 KB... Do you want it zipped or plain?

**Comment by jonasfj** [comment]
Just paste the output to a gist is fine...

**Comment by kklepper** [comment]
Oh, never did that, I'll try...

**Comment by kklepper** [comment]
Here it is:

https://gist.github.com/kklepper/413be2a85324406e50cb1362d6bec56c

**Comment by jonasfj** [comment]
Hmm, try deleting `C:\Users\Karl\AppData\Local\Pub\Cache` and downloading a
fresh flutter SDK.
It looks like something is corrupted in your PUB_CACHE, I would be curious
if that reproduces when you download again.

It's certainly interesting that `pub` installed a package, but is unable to
read the pubspec.yaml from the installed file.
There is probably something we can do in `pub` to prevent this, verifying
the download checksum, and verifying the pubspec.yaml before storing the
files in PUB_CACHE.

But I would be very curious to know if this reproduces when you clear the
cache and use a new/clean flutter-sdk.
If that reproduces it suggests that somehow files are garbled when
extracted on your system.
(the files are downloaded over HTTPS, so I think it shouldn't be possible
that they are garbled due to intermittent network errors).

**Comment by kklepper** [comment]
Ok, sounds plausible, will do soon.

**Comment by kklepper** [comment]
Removing `C:\Users\Karl\AppData\Local\Pub\Cache` did change nothing. 

Then I removed the `flutter` directory and started from fresh:

```
D:\data\programs\dev>git clone https://github.com/flutter/flutter.git -b stable
Cloning into 'flutter'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 248787 (delta 0), reused 0 (delta 0), pack-reused 248779
Receiving objects: 100% (248787/248787), 105.58 MiB | 12.05 MiB/s, done.
Resolving deltas: 100% (191115/191115), done.
Updating files: 100% (4658/4658), done.

D:\data\programs\dev>cd flutter

D:\data\programs\dev\flutter>flutter doctor
Checking Dart SDK version...
Downloading Dart SDK from Flutter engine ee76268252c22f5c11e82a7b87423ca3982e51a7...
Unzipping Dart SDK...
Building flutter tool...
Running pub upgrade...

  +----------------------------------------------------------------------------+
  ¦                 Welcome to Flutter! - https://flutter.dev                  ¦
  ¦                                                                            ¦
  ¦ The Flutter tool uses Google Analytics to anonymously report feature usage ¦
  ¦ statistics and basic crash reports. This data is used to help improve      ¦
  ¦ Flutter tools over time.                                                   ¦
  ¦                                                                            ¦
  ¦ Flutter tool analytics are not sent on the very first run. To disable      ¦
  ¦ reporting, type 'flutter config --no-analytics'. To display the current    ¦
  ¦ setting, type 'flutter config'. If you opt out of analytics, an opt-out    ¦
  ¦ event will be sent, and then no further information will be sent by the    ¦
  ¦ Flutter tool.                                                              ¦
  ¦                                                                            ¦
  ¦ By downloading the Flutter SDK, you agree to the Google Terms of Service.  ¦
  ¦ Note: The Google Privacy Policy describes how data is handled in this      ¦
  ¦ service.                                                                   ¦
  ¦                                                                            ¦
  ¦ Moreover, Flutter includes the Dart SDK, which may send usage metrics and  ¦
  ¦ crash reports to Google.                                                   ¦
  ¦                                                                            ¦
  ¦ Read about data we send with crash reports:                                ¦
  ¦ https://flutter.dev/docs/reference/crash-reporting                         ¦
  ¦                                                                            ¦
  ¦ See Google's privacy policy:                                               ¦
  ¦ https://policies.google.com/privacy                                        ¦
  +----------------------------------------------------------------------------+


Downloading Material fonts...                                       1,0s
Downloading Gradle Wrapper...                                       0,3s
Downloading package sky_engine...                                   0,6s
Downloading flutter_patched_sdk tools...                            1,2s
Downloading flutter_patched_sdk_product tools...                    1,2s
Downloading windows-x64 tools...                                    2,6s
Downloading windows-x64/font-subset tools...                        0,7s
Downloading android-arm-profile/windows-x64 tools...                0,5s
Downloading android-arm-release/windows-x64 tools...                0,6s
Downloading android-arm64-profile/windows-x64 tools...              0,5s
Downloading android-arm64-release/windows-x64 tools...              0,5s
Downloading android-x64-profile/windows-x64 tools...                0,5s
Downloading android-x64-release/windows-x64 tools...                0,5s
Doctor summary (to see all details, run flutter doctor -v):
[v] Flutter (Channel stable, v1.17.5, on Microsoft Windows [Version 10.0.19041.388], locale de-DE)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    X Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[!] Android Studio (version 4.0)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
[!] VS Code
    X Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[!] Connected device
    ! No devices available

! Doctor found issues in 4 categories.

```
1)
    X Android licenses not accepted.

```
D:\data\Programme\dev\flutter>flutter doctor --android-licenses
7 of 7 SDK package licenses not accepted. 100% Computing updates...
Review licenses that have not been accepted (y/N)? Warning: File C:\Users\Karl\.android\repositories.cfg could not be loaded.
y

[...]

Accept? (y/N): y
All SDK package licenses accepted
```
2)
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.

did that

3)
    X Flutter extension not installed

did that

4)
    ! No devices available

Well, I guess they mean my smartphone -- yes it is not connected.

Looks good to me. What do you  think?

Thank you all very much.

**Comment by kklepper** [comment]
Special thanks to @jonasfj !

**Comment by jonasfj** [comment]
Hmm, so it's interesting how the cache got corrupted the first time around :/

The fact that it's intermittent, suggests to me that we should do more to verify the downloads we do.

**Comment by kklepper** [comment]
What can I do? Let me remember.

It looks like the dart SDK had been installed before for some reason, but as flutter installs it's own, that should not matter. 

Next I started with a download of the zip file and unzipped it to some directory. I guess it was because that looked like an awful large job, I read on and found out about git clone and tried that as well which went like a breeze in comparison. 

So I had 2 directories with flutter.

I remember that these directories had different sizes, the unzipped more than 1 GB, the cloned less.

In my last attempt I used the clone version only. Why is the zipped version there in the first place? If you want to use flutter, you need git anyway.

**Comment by jonasfj** [comment]
Hmm, interesting.. I guess this could be many different things.

**Comment by kklepper** [comment]
I was tempted to use the zip version because it was offered first and it claimed to be a serious offer. The clone version came second and looked more like a poor man's solution. 

Why did they do that? If you are eager to proceed you will take the first best. I had a second look only after the download started. And then I thought that the zip version must be more authoritative, so I opted for that.

Why are both installations so different in size? What other differences are there? How can both interfere?

Now the question is if the problem I experienced can be reproduced at all. Answers to the above questions may give a hint.
